webforces/server/mongodbworker.py

```diff
@@ -312,6 +312,29 @@ def getAllAlgs(self) -> Tuple[DBStatus, List[Algorithm]]:
         logger.debug("Algorithms was found")
         return (DBStatus.s_ok, algs)
 
+    def getAllAlgsWithFlags(self, user_id) -> Tuple[DBStatus, List[Algorithm], List[bool]]:
+        try:
+            st, user = self.getUserByID(user_id)
+            if st == DBStatus.s_data_issue:
+                logger.error("This user does not exist")
+                return (DBStatus.s_data_issue, [Algorithm(ERROR_ID)], [ERROR_ID])
+            algs_collection = self.db["algs"]
+            algs_tmp = list(algs_collection.find({}))
+            algs = []
+            flags = []
+            for alg_d in algs_tmp:
+                flag = 0
+                algs.append(Algorithm.fromDict(alg_d))
+                alg = Algorithm.fromDict(alg_d)
+                if (alg.cost > 0) or (alg.alg_id in user.algs_id) or (alg.alg_id in user.bound_ids):
```

Comment:
`alg.cost > 0` isn't it already checked on addition?

Reply:
Algorithm can be free (cost == 0).

Reply:
There are needed `alg.cost == 0` to mark free algorithms as available

Reply:
It makes sense, thanks.